src/components/EditProfile.jsx

```diff
@@ -0,0 +1,214 @@
+import React, { useContext, useEffect, useState } from "react";
+import { UserContext } from "../context/UserContext";
+import { userData } from "../UserData";
+import { useNavigate, useParams } from "react-router";
+
+const EditProfile = () => {
+  const param = useParams();
+  const navigate = useNavigate();
+  const { name } = useContext(UserContext);
+  const [formInputs, setFormInputs] = useState({
+    id: "",
+    userName: "",
+    firstName: "",
+    lastName: "",
+    designation: "",
+    favoriteFood: "",
+    hobbies: "",
+  });
+
+  const currentUser = userData.find((user) => user.name === param.name);
+
+  useEffect(() => {
+    setFormInputs(currentUser);
+  }, [currentUser]);
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+
+    setFormInputs((prev) => {
+      return { ...prev, [name]: value };
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    navigate(`/${name}`);
+  };
+
+  const deleteFood = (food) => {
+    const newFoodArray = currentUser.favoriteFood.split(",");
+    const newFoodList = newFoodArray.filter(
+      (existingFood) => existingFood !== food
+    );
+    const newFoodString = newFoodList.join();
+
+    setFormInputs({ ...currentUser, favoriteFood: newFoodString });
+  };
+
+  const deleteHobbies = (deletedHobby) => {
+    const newHobbyArray = currentUser.hobbies.split(",");
+    const newHobbyList = newHobbyArray.filter(
+      (existingHobbies) => existingHobbies !== deletedHobby
+    );
+
+    const newHobbiesString = newHobbyList.join();
+
+    setFormInputs({ ...currentUser, hobbies: newHobbiesString });
+  };
+
+  return (
+    <div className="container">
+      {currentUser && (
+        <div className="profile">
+          <h1 className="title title-1 profile-name">Edit profile</h1>
+
+          <div className="edit-profile">
+            <form onSubmit={handleSubmit}>
+              <div className="edit-item">
+                <div className="form-holder">
+                  <div className="profile-banner">
+                    <img src="../5.jpg" alt="banner-pic" />
```

Comment:
I think it's better to import images from assets. Also please use absolute path (ex: assets/images/5.jpg)